### ISSUE TITLE
ignore-match import mode ignored when importing json using appconfig/kvset profile

### ISSUE THREAD

**Issue by carlin-q-scott** [Azure/AppConfiguration#919]
When I import a json file using the following parameters, the existing key-values are overwritten:

> --source file --format json --profile appconfig/kvset --import-mode ignore-match --yes

**Comment by albertofori** [comment]
Hi @carlin-q-scott , could you please provide information on the version of the cli that you are currently using?

**Comment by jimmyca15** [comment]
@carlin-q-scott 

Are you able to help with the version information @albertofori mentioned?

**Comment by jimmyca15** [comment]
Closing as Albert has provided an explanation on the expected behavior and there is no response.

**Comment by carlin-q-scott** [comment]
@albertofori 
> we only avoid overwriting existing key-values in ignore-match mode that have not changed. With ignore-match, a key-value with a given key and label will only be ignored if the value and content-type in your configuration file are identical to what is already in the server.

I don't understand why that would be a useful feature. Only ignore the match if nothing has changed at all? Shouldn't it always do that regardless of CLI flags? I don't care if you overwrite an identical value, but that is less efficient in the backend, so that seems like an internal implementation detail that shouldn't be exposed through the public CLI.

The CLI help text also contradicts your statement:
> If import mode is "ignore-match", source key-values **that already exist at the destination will not be overwritten**. Import mode "all" writes all key-values to the destination regardless of whether they exist or not.  Allowed values: all, ignore-match.  Default: ignore-match.

It's clearly stating that won't overwrite existing keys.

**Comment by albertofori** [comment]
@carlin-q-scott May I know how you are identifying that an existing key-value that has not changed in your json file is being overwritten? The only way I see you could tell would be using the key-value's `etag` or `last_modified` property. Perhaps, some examples of this behavior with dummy data would be of help here. 

> Shouldn't it always do that regardless of CLI flags? I don't care if you overwrite an identical value, but that is less efficient in the backend, so that seems like an internal implementation detail that shouldn't be exposed through the public CLI.

I agree with this statement and that is why we default to `ignore-match` in order to avoid unnecessary writes and consumption of write request quota. However, this has not always been the behavior of the CLI. Prior to the introduction of this behavior, all key-values in a configuration file used to be written during import regardless of whether they have changed or not. On every write, some internally set properties always change, i.e., etag and last_modified, and some users might be relying on these properties to tell whether a recent import has been made to their resource. This made it necessary to keep the original behavior as an option, i.e., `import-mode --all`.

> Only ignore the match if nothing has changed at all?

If none of the properties of a `given key-value` has changed. (i.e., key, label, content-type, value, tags), we do not write said key-value. Just to clarify, I am not talking about the whole configuration file here; comparison is done for each key-value in the file. Also note that a key-value is uniquely identified by the key and label properties. So if you update the label or key of a configuration in your kvset file, this is considered a new configuration setting altogether and is written as such.

**Comment by carlin-q-scott** [comment]
Thank you @albertofori for clarifying why the default of overwriting existing key-values was preserved. I appreciate the complexity in supporting potential use cases of a public API.

In regard to your question, I am not modifying the label of the keys that I am importing. In fact, I'm not modifying the values either; Our operations engineer is doing that in through App Config GUI. On the next deployment through our CD system, we are re-importing the same key-label-values as we did when the keys were created initially. The idea is that the import will ignore matching key-labels and only add new ones requested by our development team by adding entries to the kvset.

The reason I chose Azure App Config over a different kv-store, is that it has a pretty good GUI integrated with Azure Portal that can be used to modify values after creation. I think it loses value as a solution if there isn't a create-only kv-import option through the CLI or API.

I also still think the implementation of ignore-match import mode contradicts the help text. The key-values do already exist and are being overwritten.
